content/transfer-fee.md

```diff
@@ -0,0 +1,959 @@
+
+```ts
+/**
+ * Assign a new authority to the account
+ *
+ * @param connection       Connection to use
+ * @param payer            Payer of the transaction fees
+ * @param account          Address of the account
+ * @param currentAuthority Current authority of the specified type
+ * @param authorityType    Type of authority to set
+ * @param newAuthority     New authority of the account
+ * @param multiSigners     Signing accounts if `currentAuthority` is a multisig
+ * @param confirmOptions   Options for confirming the transaction
+ * @param programId        SPL Token program account
+ *
+ * @return Signature of the confirmed transaction
+ */
+
+await setAuthority(
+  connection,
+  payer,
+  mint,
+  currentAuthority, 
+  AuthorityType.TransferFeeConfig, // or AuthorityType.WithheldWithdraw
+  newAuthority, 
+  [],
+  undefined,
+  TOKEN_2022_PROGRAM_ID
+)
+```
+
+# Lab
+
+In this lab, we are going to create a transfer fee configured mint. We'll use a fee vault to hold the transfer fees, and we'll collect the fees using both the direct and the harvesting methods.
+
+### 1. Getting started
+To get started, clone [this repository's](https://github.com/Unboxed-Software/solana-lab-transfer-fee.git) `starter` branch.
+
+```bash
+git clone https://github.com/Unboxed-Software/solana-lab-transfer-fee.git
+cd solana-lab-transfer-fee
+git checkout starter
+npm install
+```
+
+The starter code comes with following file:
+ - `index.ts`
+
+`index.ts` has a main function that creates a connection to the specified cluster and calls `initializeKeypair`. This `main` function is where we'll be writing our script.
+
+### 2. Run validator node
+
+For the sake of this guide, we'll be running our own validator node.
+
+In a separate terminal, run the following command: `solana-test-validator`. This will run the node and also log out some keys and values. The value we need to retrieve and use in our connection is the JSON RPC URL, which in this case is `http://127.0.0.1:8899`. We then use that in the connection to specify to use the local RPC URL.
+
+```tsx
+const connection = new Connection("http://127.0.0.1:8899", "confirmed");
+```
+
+Alternatively, if you’d like to use testnet or devnet, import the `clusterApiUrl` from `@solana/web3.js` and pass it to the connection as such:
+
+```tsx
+const connection = new Connection(clusterApiUrl('devnet'), 'confirmed');
+```
+
+If you decide to use devnet, and have issues with airdropping SOL. Feel free to add the `keypairPath` parameter to `initializeKeypair`. You can get this from running `solana config get` in your terminal. And then go to [faucet.solana.com](https://faucet.solana.com/) and airdrop some sol to your address. You can get your address from running `solana address` in your terminal.
```

Comment:
I'd just use:

```
const newBalance = await airdropIfRequired(
  connection,
  keypair.publicKey,
  0.5 * LAMPORTS_PER_SOL,
  1 * LAMPORTS_PER_SOL,
);
```

https://github.com/Unboxed-Software/solana-lab-transfer-fee/blob/main/src/keypair-helpers.ts seems to duplicate what's already in the helpers library, and we want users to be able to create programs from scratch using well known libraries. There's around a thousand downloads a week for the helpers library right now.